### PR TITLE
another pass at linting

### PR DESCRIPTION
**Pull request by zingale**
## PR checklist

- [ ] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite to more than roundoff level
- [ ] all newly-added functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated, if appropriate
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

File: Source/driver/Castro.cpp [hunk 1]
```diff
@@ -146,32 +146,33 @@ void
 Castro::variableCleanUp ()
 {
 #ifdef GRAVITY
-  if (gravity != 0) {
-    if (verbose > 1 && ParallelDescriptor::IOProcessor()) {
+  if (gravity != nullptr) {
+    if (verbose >= 1 && ParallelDescriptor::IOProcessor()) {
```
Comment (maxpkatz) [inline Source/driver/Castro.cpp:150]: I think `verbose > 1` was correct here, this is too much information for the typical user.

File: Source/driver/Castro.cpp [hunk 2]
```diff
@@ -146,32 +146,33 @@ void
...
     delete diffusion;
-    diffusion = 0;
+    diffusion = nullptr;
   }
 #endif
 
 #ifdef RADIATION
-  if (radiation != 0) { int report = (verbose || radiation->verbose);
+  if (radiation != nullptr) {
+    int report = (verbose >= 1 || radiation->verbose);
```
Comment (maxpkatz) [inline Source/driver/Castro.cpp:170]: Let's also make these both `verbose > 1`

File: Source/driver/Castro.cpp [hunk 3]
```diff
@@ -1353,7 +1371,7 @@ Castro::init ()
     // If we just triggered a regrid, we need to account for the fact that
     // the data on the coarse level has already been advanced.
 
-    if (getLevel(level-1).post_step_regrid)
+    if (getLevel(level-1).post_step_regrid = 1)
```
Comment (maxpkatz) [inline Source/driver/Castro.cpp:1374]: ==

**Comment by maxpkatz** [comment]
For all the places where this PR does e.g. `if (do_grav)` -> `if (do_grav == 1)`, then to keep the meaning the same I think we should either test on `if (do_grav != 0)` or add a check at initialization that aborts if the value is anything other than 0 or 1.

**Comment by zingale** [comment]
this PR is out of date

**Pull request closed without merging**